ts/src/zebpayfutures.ts

```diff
@@ -0,0 +1,1310 @@
+     */
+    async createOrder (symbol: string, type: OrderType, side: OrderSide, amount: number, price: Num = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const marginAsset = this.safeString (params, 'marginAsset', 'INR');
+        const leverage = this.safeString (params, 'leverage');
+        if (leverage === undefined) {
+            throw new ArgumentsRequired (this.id + ' createOrder() requires a leverage parameter argument');
+        }
+        if (type !== 'limit' && type !== 'market') {
+            throw new BadRequest (`${this.id} createOrder() type must be either 'market' or 'limit'`);
+        }
+        const formType = this.safeString (params, 'formType', 'ORDER_FORM');
+        const upperCaseFormType = formType.toUpperCase ();
+        const upperCaseType = type.toUpperCase ();
+        const takeProfit = this.safeBool (params, 'takeProfit', false);
+        const stopLoss = this.safeBool (params, 'stopLoss', false);
+        const timestamp = this.safeInteger (params, 'timestamp');
+        const orderType = this.safeString (params, 'orderType');
+        const positionId = this.safeString (params, 'positionId', undefined);
+        params = this.omit (params, [ 'marginAsset', 'leverage', 'formType', 'positionId', 'orderType' ]);
+        const request: Dict = {
+            'formType': upperCaseFormType,
+            'amount': amount,
+            'side': side,
+            'marginAsset': marginAsset,
+            'leverage': leverage,
+            'symbol': market['id'],
+            'type': upperCaseType, // limit or market
+            'timestamp': timestamp,
+        };
+        if (type === 'limit') {
+            if (price === undefined) {
+                throw new ArgumentsRequired (this.id + ' createOrder() requires a price argument for limit orders');
+            } else {
+                request['price'] = price;
+            }
+        }
+        if (takeProfit === true) {
+            request['takeProfitPrice'] = price;
+            request['positionId'] = positionId;
+            request['orderType'] = orderType;
+        }
+        if (stopLoss === true) {
+            request['stopLossPrice'] = price;
+            request['positionId'] = positionId;
+            request['orderType'] = orderType;
+        }
+        params = this.omit (params, [ 'price' ]);
+        let response = undefined;
+        if (takeProfit === true || stopLoss === true) {
+            response = await this.privatePostTradeOrderAddTPSL (this.extend (request, params));
+        } else {
+            response = await this.privatePostTradeOrder (this.extend (request, params));
+        }
+        //
+        //    {
+        //        "data": {
+        //            "clientOrderId": "619717484f1d010001510cde",
+        //        },
+        //    }
+        //
+        const data = this.safeValue (response, 'data', {});
+        return this.safeOrder ({
```

Comment:
we should use `parseOrder` (the request and response parsing  need to be detached because we need to parse orders/trades/etc in multiple places)